### DISCUSSION TITLE
What would it take for webpack to not be boilerplate anymore?

### DISCUSSION THREAD

**Discussion by airbr** [h5bp/html5-boilerplate#3069] [Ideas]
I love v9.0 release of this project but I wanted to ask if there was a plan or discussion somewhere about eventually removing webpack. It is notable that this project is still using npm and a bundler like webpack. It is an improvement on replacing Parcel but I was curious if there was a discussion like "what would it take for webpack to be removed". Improvements in HTTP? Networking? Other things? Just a thought I had.

Thanks for all the work that went into this project by all the contributors. I find it useful.

**Comment by roblarsen** [comment]
I did the WebPack implementation here and it kind of drove me crazy, so maybe just me getting annoyed at it just that little bit more is what it will take 🤣

Jokes aside, running a development server (one of the most important use cases for us to use WebPack) is always going to require _some_ 3rd party code.  I don't know that WebPack (and npm) is the best possible solution for that problem, but it is a solution and is one that people know. 

I do think the packaging of artifacts for the web could be improved at the infrastructure/standards level to make some of the features of WebPAck for packaging content  redundant. That's happened repeatedly over the past 20 or so years as we've become more and more interested in web performance. 

I may be reading your question wrong. What is it you envision?

**Comment by tuta-amb** [comment]
I personally don't use webpack anymore in my projects as it's pretty finicky (and slow!) to work with. There's quite a few newer packages, like ESBuild, SWC, Vite, and Parcel to name a few.

**Reply by roblarsen** [reply]
We removed Parcel because it simply couldn't do the things we required (like... not renaming files that need to have standard/non-cache busted names.) I looked at Vite and... said no to that for some reason. I can look it up (all of this discussion is in the issues)

ESBuild, just by the name indicates to me that it may not suit our purposes. I will look at it though. As well as SWC. 

Webpack IS finicky! I hated working in it! Most people just use it as part of a library or framework and they think it's magic. It's super weird! I'm not a fan, but it actually does what we need it to do so I went with it.

**Reply by tuta-amb** [reply]
ESBuild is a general purpose bundler. It's a lot less configurable then Webpack, however most things are either pre-setup or easily changeable.

 I personally don't use Parcel, so I can't say much about it.

And as for webpack, there was higher ups at one place I did work at, which wanted everything to be webpack (idk why). DevOps took 6 weeks to get everything working again lol
